asyncpg/protocol/record/recordobj.c

```diff
@@ -658,11 +661,69 @@ static PyMappingMethods record_as_mapping = {
 };
 
 
+static PyObject *
+record_reduce(ApgRecordObject *o, PyObject *Py_UNUSED(unused))
+{
+    PyObject *value = PyTuple_New(2);
+     if (value == NULL) {
+        return NULL;
+    }
+	Py_ssize_t len = Py_SIZE(o);
+    PyObject *state = PyTuple_New(1 + len);
+    if (state == NULL) {
+    	Py_DECREF(value);
+        return NULL;
+    }
+    PyTuple_SET_ITEM(value, 0, record_reconstruct_obj);
+    Py_INCREF(record_reconstruct_obj);
+    PyTuple_SET_ITEM(value, 1, state);
+    PyTuple_SET_ITEM(state, 0, (PyObject *)o->desc);
+    Py_INCREF(o->desc);
+    for (Py_ssize_t i = 0; i < len; i++) {
+        PyObject *item = ApgRecord_GET_ITEM(o, i);
+        PyTuple_SET_ITEM(state, i + 1, item);
+		Py_INCREF(item);
+    }
+    return value;
+}
+
+static PyObject *
+record_reconstruct(PyObject *Py_UNUSED(unused), PyObject *args)
+{
+    if (!PyTuple_CheckExact(args)) {
+        return NULL;
+    }
+    Py_ssize_t len = PyTuple_GET_SIZE(args);
+    if (len < 2) {
+        return NULL;
+    }
+    len--;
+    ApgRecordDescObject *desc = (ApgRecordDescObject *)PyTuple_GET_ITEM(args, 0);
+    if (!ApgRecordDesc_CheckExact(desc)) {
+        return NULL;
+    }
+    if (PyObject_Length(desc->mapping) != len) {
+        return NULL;
+    }
+    PyObject *record = ApgRecord_New(&ApgRecord_Type, (PyObject *)desc, len);
```

Comment:
This would lose the original record subtype if one was used, so you need to pickle the class reference also.